doc/reference/networking/openthread_integration.rst

```diff
@@ -0,0 +1,172 @@
+.. _openthread_integration:
+
+OpenThread Integration
+######################
+
+.. contents::
+    :local:
+    :depth: 2
+
+Overview
+********
+The OpenThread stack is integrated with the Zephyr RTOS as the L2 layer.
+The main advantage of this approach is that it can utilize the Zephyr's L3 layer.
+The drawback, on the other hand, lies in the reception path in which the IP packet
+needs to traverse the OpenThread L3 layer and then Zephyr's L3 layer in order to 
+reach the BSD socket.
+
+Note that if one desires the application can use just the OpenThread API 
+with it's IPv6 stack and even the internal CoAP implementation 
+which is the application protocol.
+
+File system and shim layer
+**************************
+
+The OpenThread Network Stack is present in the following paths:
+- OpenThread stack location: modules/lib/openthread/
+- OpenThread shim layer location: zephyr/subsys/net/lib/openthread/platform/
+It's responsibilities include:
+- translating the data into the Zephyr's native net_pkt,
+- providing openthread thread body and synchronization API,
+- providinng openthread_send(), openthread_recv() calls which are registered as the L2 interface API,
+- providing a way to initialize the OpenThread stack,
+- implementing callback functions used by the OpenThread stack
+
+The nRF IEEE802.15.4 Radio Driver is present in the following paths:
+- nRF IEEE802.15.4 Radio Driver shim layer location: zephyr/drivers/ieee802154/{ieee802154_nrf5.c/ieee802154_nrf5.h
+- nRF IEEE802.15.4 Radio Driver location: modules/hal/nordic/drivers/nrf_radio_802154
```

Comment:
Note: If this document is part of zephyr it's ok. If it will be a part of NCS the location will change soon to somewhere in nrf/nrfx

Reply:
Yes, we plan to integrate it with our Zephyr fork and then upstream it as well.